### PR TITLE
Fix deprication warnings

### PR DESCRIPTION
**Pull request by carlisom**
* Fix warning about not using a PEP 517 installer
  * move to pytest
  * add python 3.9 and 3.10 to doc testing
* Fix depreciation warning from using description-file vs description_file
* Fix depreciation warning from sessions to use allow_list

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -80,10 +80,10 @@ release: dist ## package and upload a release
 	twine upload --skip-existing dist/alertlogic-sdk-python-*.* dist/*
 
 dist: clean ## builds source and wheel package
-	$(PYTHON) setup.py sdist
+	$(PYTHON) -m build
 
-install: clean ## install the package to the active Python's site-packages
-	$(PYTHON) setup.py install
+install: ## install the package to the active Python's site-packages
+	$(PIP) install pip install dist/alertlogic-sdk-python*.tar.gz
```
Comment (james-bellamy) [inline Makefile:86]: To remove all dep warnings I think this should be  `$(PYTHON) -m pip install dist/alertlogic-sdk-python*.tar.gz` as pip doesn't like being called directly anymore

File: requirements_dev.txt [hunk 2]
```diff
@@ -1,18 +1,21 @@
...
+twine>=1.13.0
+configparser>=4.0.2
+sphinx_rtd_theme>=0.4.3
+sphinxcontrib_contentui>=0.2.4
+sphinx-paramlinks>=0.5.4
+m2r2>=0.3.2
 boto3>=1.16.57
+pytest>=6.2.5
+build>=0.9.0
+alertlogic-sdk-definitions>=0.1.96
```
[the remark this answers is not in the document]
Comment (carlisom) [reply requirements_dev.txt:21]: The latest of this of this PR

File: Makefile (hunk 1 above)
Comment (carlisom) [reply Makefile:86]: Its more of a best practice I see as it does not throw a warning, will update it though to follow the best practices.

Comment (carlisom) [reply Makefile:86]: Addressed via https://github.com/alertlogic/alertlogic-sdk-python/pull/114/commits/b60ce052216d260db89ac30baf384b4e369a59a7

**Review by james-bellamy — APPROVED** [review]

**Pull request merged**